provisioner/powershell/provisioner_acc_test.go

```diff
@@ -1,94 +1,110 @@
 package powershell_test
 
 import (
-	"bytes"
 	"fmt"
 	"io/ioutil"
 	"os"
+	"os/exec"
 	"path/filepath"
+	"runtime"
 	"testing"
 
-	"github.com/hashicorp/go-uuid"
-	"github.com/hashicorp/packer/command"
 	"github.com/hashicorp/packer/packer-plugin-sdk/acctest/provisioneracc"
-	packersdk "github.com/hashicorp/packer/packer-plugin-sdk/packer"
-	"github.com/hashicorp/packer/provisioner/powershell"
-	windowsshellprovisioner "github.com/hashicorp/packer/provisioner/windows-shell"
 )
 
-const TestProvisionerName = "powershell"
+const TestProvisionerType = "powershell"
 
-func TestAccPowershellProvisioner_basic(t *testing.T) {
-	provisioneracc.TestProvisionersPreCheck(TestProvisionerName, t)
-
-	testProvisioner := PowershellProvisionerAccTest{"powershell-provisioner-cleanup.txt"}
-	provisioneracc.TestProvisionersAgainstBuilders(&testProvisioner, t)
+func powershellIsCompatible(builder string, vmOS string) bool {
+	return vmOS == "windows"
 }
 
-func TestAccPowershellProvisioner_Inline(t *testing.T) {
-	provisioneracc.TestProvisionersPreCheck(TestProvisionerName, t)
-
-	testProvisioner := PowershellProvisionerAccTest{"powershell-inline-provisioner.txt"}
-	provisioneracc.TestProvisionersAgainstBuilders(&testProvisioner, t)
+func fixtureDir() string {
+	_, file, _, _ := runtime.Caller(0)
+	return filepath.Join(filepath.Dir(file), "test-fixtures")
 }
 
-func TestAccPowershellProvisioner_Script(t *testing.T) {
-	provisioneracc.TestProvisionersPreCheck(TestProvisionerName, t)
-
-	testProvisioner := PowershellProvisionerAccTest{"powershell-script-provisioner.txt"}
-	provisioneracc.TestProvisionersAgainstBuilders(&testProvisioner, t)
-}
+func LoadProvisionerFragment(templateFragmentPath string) (string, error) {
+	dir := fixtureDir()
+	fragmentAbsPath := filepath.Join(dir, templateFragmentPath)
+	fragmentFile, err := os.Open(fragmentAbsPath)
+	if err != nil {
+		return "", fmt.Errorf("Unable find %s", fragmentAbsPath)
+	}
+	defer fragmentFile.Close()
 
-type PowershellProvisionerAccTest struct {
-	ConfigName string
-}
+	fragmentString, err := ioutil.ReadAll(fragmentFile)
+	if err != nil {
+		return "", fmt.Errorf("Unable to read %s", fragmentAbsPath)
+	}
 
-func (s *PowershellProvisionerAccTest) GetName() string {
-	return TestProvisionerName
+	return string(fragmentString), nil
 }
 
-func (s *PowershellProvisionerAccTest) GetConfig() (string, error) {
-	filePath := filepath.Join("./test-fixtures", s.ConfigName)
-	config, err := os.Open(filePath)
+func TestAccPowershellProvisioner_basic(t *testing.T) {
```

Comment:
I like the way it looks now!

Reply:
thanks! I kind of copied the builder testcase which means the builder and provisioner acceptance tests now kinda look the same and I think that's nice for us as maintainers.